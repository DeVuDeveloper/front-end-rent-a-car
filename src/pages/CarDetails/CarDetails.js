import { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { useParams, NavLink } from 'react-router-dom';
import { RiArrowRightSFill } from 'react-icons/ri';
import { HiChevronDoubleRight } from 'react-icons/hi';
import { carId } from '../../redux/reducers/reservations/carId';
import style from './Detail.module.scss';

const CarDetails = () => {
  const { Id } = useParams();
  const [car, setCar] = useState({});
  const fetchCar = async () => {
    const res = await fetch(`https://final-capstone-back.herokuapp.com/api/v1/cars/${Id}`, {
      headers: {
        'Content-Type': 'application/json',
        Authorization: localStorage.getItem('token'),
      },
    });
    const data = await res.json();
    setCar(data);
  };
  useEffect(() => {
    fetchCar();
  }, []);

  const dispatch = useDispatch();

  const setCarId = () => {
    dispatch(carId(car.id));
  };

  return (
    <section className={style.show_container}>
      <div className={style.show_img}>
        <div className={style.img_div}>
          <img alt="Car" src={car.photo} />
        </div>
      </div>

      <div className={style.show_content}>
        <div className={style.show_header}>
          <h2>{car.car_model}</h2>
          <p>-$200 deposit upon any Rental Purchase!</p>
        </div>

        <div className={style.show_details}>
          <div className={style.car_details}>
            <p className={style.details}>{car.description}</p>
          </div>
          <div className={style.car_detail}>
            <p className={style.details}>Rent daily cost</p>
            <p className={style.details}>
              $
              {
                car.price_per_day
                }
            </p>
          </div>
          <div className={style.car_details}>
            <p className={style.details}>Car type</p>
            <p className={style.details}>{car.car_type}</p>
          </div>
          <div className={style.car_detail}>
            <p className={style.details}>Transmission</p>
            <p className={style.details}>{car.transmission}</p>
          </div>
          <div className={style.show_discount}>
            <p>
              <span className={style.dis_amt}>
                {car.price_per_day / 10}
                % APR
              </span>
              <span className={style.dis_text}>representative</span>
            </p>
          </div>
        </div>

        <div className={style.more_cars}>
          <NavLink
            to={{
              pathname: '/home',
            }}
            className={style.d_flex}
          >
            <span className={style.discover}>DISCOVER MORE CARS</span>
            <RiArrowRightSFill className={style.success} style={{ fontSize: 40 }} />
          </NavLink>
        </div>

        <NavLink
          to={{
            pathname: `/add_reservations/${car.id}`,
          }}
          className={style.btn_container}
        >
          <button
            type="button"
            className={style.btn}
            onClick={
                 () => setCarId(car.id)
                }
          >
            <span>Reserve</span>
            {' '}
            <HiChevronDoubleRight style={{ fontSize: 30 }} />
          </button>
        </NavLink>
      </div>
    </section>
  );
};

export default CarDetails;
