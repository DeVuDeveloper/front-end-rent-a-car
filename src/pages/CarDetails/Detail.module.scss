.car_details_page {
  height: 100vh;
  display: flex;
  flex-direction: column;
  padding-top: 4rem;

  .header {
    z-index: 999;
  }

  .sidebar {
    display: none;
  }
}

.show_container {
  position: relative;
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  width: 100%;

  .show_img {
    background-color: transparent;
    height: 30vh;

    img {
      max-height: 250px;
      margin: 0 auto;
      outline: none;
      background-color: transparent;
    }

    .btn_home {
      align-self: flex-start;
    }
  }

  .btn_container {
    align-self: flex-end;
    margin-right: 5%;
    margin-top: 50px;

    .btn {
      display: flex;
      align-self: center;
      background-color: #97bf0f;
      padding: 1rem 2rem;
      border-radius: 10rem;
      color: #fff;
      text-transform: uppercase;
      font-size: 1.2rem;
      font-weight: bold;
      font-family: 'Lato', sans-serif;
      letter-spacing: 0.15rem;
      transition: all 0.3s;
      position: relative;
      overflow: hidden;
      z-index: 1;
      border: none;
      cursor: pointer;

      &:hover {
        color: #fff;

        &::before {
          width: 100%;
        }
      }
    }
  }

  .show_content {
    display: flex;
    flex-direction: column;
    align-items: center;

    .show_header {
      display: flex;
      flex-direction: column;
      width: 100%;
      align-items: center;

      h2 {
        font-size: 2rem;
        font-weight: bold;
        text-align: end;
        text-transform: uppercase;
        font-family: 'Lato', sans-serif;
      }

      p {
        font-size: 1.2rem;
        font-weight: bold;
        text-align: end;
        font-family: 'Lato', sans-serif;
      }
    }

    .show_details {
      width: 90%;

      .car_details {
        display: flex;
        justify-content: space-between;
        background-color: lightgrey;
        font-weight: 400;

        .details {
          font-size: 1.2rem;
          font-family: 'Lato', sans-serif;
          font-weight: 400;
          margin: 0;
          padding: 8px 3px;
        }
      }

      .car_detail {
        display: flex;
        justify-content: space-between;

        .details {
          font-size: 1.2rem;
          font-family: 'Lato', sans-serif;
          font-weight: 400;
          padding: 8px 3px;
          margin: 0;
        }
      }

      .show_discount {
        margin-top: 5px;

        p {
          padding: 0 15px;
          display: flex;
          gap: 0.5rem;
          font-size: 1.4rem;

          .dis_amt {
            text-transform: uppercase;
            font-weight: bold;
            font-family: 'Lato', sans-serif;
          }

          .dis_text {
            text-transform: capitalize;
            font-family: 'Lato', sans-serif;
          }
        }
      }
    }

    .more_cars {
      margin-right: 5%;
      align-self: flex-end;
      margin-top: 1rem;
      display: flex;
      align-items: flex-end;
      justify-content: center;

      span {
        font-weight: bolder;
        font-size: 1.2rem;
        padding-bottom: 4px;
        justify-self: center;
        align-self: center;
        font-family: 'Lato', sans-serif;
      }
    }
  }
}

.success {
  color: #42f10c;
}

@media screen and (min-width: 768px) and (max-width: 1300px) {
  .car_details_page {
    display: flex;
    flex-direction: row;
    padding-top: 0 !important;

    .sidebar {
      display: flex;
    }
  }

  .btn_container {
    margin-bottom: 30px !important;
  }

  .details {
    padding: 8px 13px !important;
  }
}

@media screen and (min-width: 1301px) {
  .car_details_page {
    display: flex;
    flex-direction: row;
    padding-top: 0 !important;

    .sidebar {
      display: flex;
      width: 30%;
    }
  }

  .show_container {
    display: flex;
    flex-direction: row;
    align-items: center;

    .show_img {
      width: 50%;
    }

    .show_content {
      width: 50%;
    }

    .btn_container {
      margin-top: 70px;
    }
  }
}

.d_flex {
  display: flex;
}
