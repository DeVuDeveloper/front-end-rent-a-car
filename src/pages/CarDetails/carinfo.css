.car-details {
  display: flex;
}

.car-info {
  display: flex;
  align-self: center;
  gap: 50px;
  margin: auto;
  padding: 100px;
  width: 80%;
}

.car-photo {
  width: 100%;
  height: 100%;
}

.infos {
  padding: 0;
  margin: 0 0 100px 0;
}

.infos li {
  display: flex;
  justify-content: space-between;
  padding: 0;
  margin-bottom: 0;
  width: 100%;
  align-items: center;
}

.infos li:nth-child(odd) {
  background-color: rgb(221, 214, 214);
}

.discount p {
  font-weight: bolder;
}

.discount span {
  font-weight: 300;
}

.infos li p {
  padding: 0 10px;
  margin-bottom: 0;
  font-family: 'Lato', sans-serif;
  font-size: 14px;
  color: rgb(78, 77, 74);
}

.reserve-button {
  padding: 3px 17px;
  background-color: rgb(152, 192, 42);
  border: none;
  border-radius: 30px;
  color: #fff;
  display: flex;
  align-items: center;
  gap: 5px;
}

.reserve-button h3 {
  padding: 0;
  margin: 0;
  color: #fff;
}

@media screen and (max-width: 1243px) {
  .navbar {
    display: none;
  }

  .car-info {
    display: flex;
    flex-direction: column;
    align-self: center;
    gap: 10px;
    margin: auto;
    padding: 20px;
    width: 100%;
    margin-top: 80px;
  }

  .reserve-button {
    padding: 3px 17px;
    background-color: rgb(152, 192, 42);
    border: none;
    border-radius: 30px;
    color: #fff;
    display: flex;
    align-items: center;
    gap: 5px;
    margin-top: -60px;
  }

  .column {
    padding: 0;
    margin: 0;
  }
}
