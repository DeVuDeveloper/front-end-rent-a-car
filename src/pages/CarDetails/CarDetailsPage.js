import React from 'react';
import CarDetails from './CarDetails';
import Sidebar from '../../components/navigation/Sidebar';
import Hamburger from '../../components/navigation/Hamburger';
import style from './Detail.module.scss';

const CarDetailsPage = () => (
  <section className={style.car_details_page}>
    <div className={style.header}>
      <Hamburger />
    </div>
    <div className={style.sidebar}>
      <Sidebar />
    </div>
    <CarDetails />
  </section>
);

export default CarDetailsPage;
