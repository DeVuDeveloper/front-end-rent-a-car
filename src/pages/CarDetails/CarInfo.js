import React, { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { useParams, NavLink } from 'react-router-dom';
import { BiFastForwardCircle } from 'react-icons/bi';
import { carId } from '../../redux/reducers/reservations/carId';
import Sidebar from '../../components/navigation/Sidebar';
import Hamburger from '../../components/navigation/Hamburger';

import './carinfo.css';

const CarDetails = () => {
  const { Id } = useParams();
  const [car, setCar] = useState({});
  const fetchCar = async () => {
    const res = await fetch(`https://final-capstone-back.herokuapp.com/api/v1/cars/${Id}`, {
      headers: {
        'Content-Type': 'application/json',
        Authorization: localStorage.getItem('token'),
      },
    });
    const data = await res.json();
    setCar(data);
  };
  useEffect(() => {
    fetchCar();
  }, []);

  const dispatch = useDispatch();

  const setCarId = () => {
    dispatch(carId(car.id));
  };

  return (
    <section className="car-details">
      <Hamburger pageWrapId="page-wrap" outerContainerId="outer-container" />
      <div className="navbar">
        <Sidebar />
      </div>
      <div className="car-info" id={carId}>
        <div className="car-image">
          <img className="car-photo" src={car.photo} alt="" />
        </div>
        <div className="info-wrapper">
          <div className="column">
            <div className="car-info-name">
              <h2 className="rocket-name">{car.car_model}</h2>
              <p>-$250 deposit upon any Rental Purchase!</p>
            </div>
            <ul className="infos">
              <li className="rocket-name">
                <p>price for day:</p>
                <p>{car.price_per_day}</p>
              </li>
              <li className="rocket-name">
                <p>Car type:</p>
                <p>{car.car_type}</p>
              </li>
            
              <li className="discount">
                <p>
                  5.9% APR-
                  <span>Discount</span>
                </p>
              </li>
            </ul>
          </div>
          <div className="button">
            <NavLink to="/add_reservations">
              <button
                className="reserve-button"
                type="button"
                onClick={() => setCarId(car.id)}
              >
                <h3>Reserve</h3>
                <BiFastForwardCircle />
              </button>
            </NavLink>
          </div>
        </div>
      </div>
    </section>
  );
};

export default CarDetails;
