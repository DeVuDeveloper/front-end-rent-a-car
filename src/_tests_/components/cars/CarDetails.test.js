import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CarDetails from '../../../pages/CarDetails/CarDetails';
import store from '../../../redux/configureStore';

it('should render CarDetails component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <CarDetails />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
