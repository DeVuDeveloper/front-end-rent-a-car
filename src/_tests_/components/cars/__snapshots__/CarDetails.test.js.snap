// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CarDetails component correctly 1`] = `
<section
  className="show_container"
>
  <div
    className="show_img"
  >
    <div
      className="img_div"
    >
      <img
        alt="Car"
      />
    </div>
  </div>
  <div
    className="show_content"
  >
    <div
      className="show_header"
    >
      <h2 />
      <p>
        -$200 deposit upon any Rental Purchase!
      </p>
    </div>
    <div
      className="show_details"
    >
      <div
        className="car_details"
      >
        <p
          className="details"
        />
      </div>
      <div
        className="car_detail"
      >
        <p
          className="details"
        >
          Rent daily cost
        </p>
        <p
          className="details"
        >
          $
        </p>
      </div>
      <div
        className="car_details"
      >
        <p
          className="details"
        >
          Car type
        </p>
        <p
          className="details"
        />
      </div>
      <div
        className="car_detail"
      >
        <p
          className="details"
        >
          Transmission
        </p>
        <p
          className="details"
        />
      </div>
      <div
        className="show_discount"
      >
        <p>
          <span
            className="dis_amt"
          >
            NaN
            % APR
          </span>
          <span
            className="dis_text"
          >
            representative
          </span>
        </p>
      </div>
    </div>
    <div
      className="more_cars"
    >
      <a
        aria-current={null}
        className="d_flex"
        href="/home"
        onClick={[Function]}
      >
        <span
          className="discover"
        >
          DISCOVER MORE CARS
        </span>
        <svg
          className="success"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
              "fontSize": 40,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16 12l-6 6V6z"
            />
          </g>
        </svg>
      </a>
    </div>
    <a
      aria-current={null}
      className="btn_container"
      href="/add_reservations/undefined"
      onClick={[Function]}
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        <span>
          Reserve
        </span>
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
              "fontSize": 30,
            }
          }
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10.293 15.707a1 1 0 010-1.414L14.586 10l-4.293-4.293a1 1 0 111.414-1.414l5 5a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M4.293 15.707a1 1 0 010-1.414L8.586 10 4.293 5.707a1 1 0 011.414-1.414l5 5a1 1 0 010 1.414l-5 5a1 1 0 01-1.414 0z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </a>
  </div>
</section>
`;
